### PR TITLE
add `reply()` for `poise::context`

### PR DESCRIPTION
**Pull request by B-2U**
the only problem is the name, `reply_reply` is definitly confusing, but I don't have a better idea than it :(

### CODE REVIEW

File: src/reply/send_reply.rs [hunk 1]
```diff
@@ -44,6 +44,18 @@ pub async fn say_reply<U, E>(
     send_reply(ctx, |m| m.content(text.into())).await
 }
 
+/// Shorthand of [`send_reply`] for text-only messages, and reply to the reference message for [`crate::PrefixContext`]
+///
+/// for [`crate::ApplicationContext`], this is the same as [`say_reply`].
+///
+/// Note: panics when called in an autocomplete context!
+pub async fn reply_reply<U, E>(
```
Comment (kangalio) [inline src/reply/send_reply.rs]: I say remove `reply_reply` - those plain functions for replying have become kinda useless anyways (`say_reply(ctx, "...")` is inferior to `ctx.say("...")`) so it's not really important to add this new utility feature to them

File: src/structs/context.rs [hunk 2]
```diff
@@ -131,6 +131,21 @@ context_methods! {
...
     }
 
+    /// Shorthand of [`crate::reply_reply`]
+    ///
+    /// reply to the reference message in prefix commands
+    ///
+    /// for slash commands, this is the same as [`say`].
+    ///
+    /// Note: panics when called in an autocomplete context!
+    await (say self text)
```
Comment (kangalio) [inline src/structs/context.rs]: Please change the method name here from say to reply, or the method implementations on PrefixContext and ApplicationContext will have the wrong name

**Review by kangalio — CHANGES_REQUESTED** [review]

File: src/reply/send_reply.rs (hunk 1 above)
Comment (B-2U) [reply src/reply/send_reply.rs]: do you mean I should directly use `send_reply()` builder in `ctx.reply()` ?

Comment (kangalio) [reply src/reply/send_reply.rs]: Yes, or even `ctx.send(|b| b.content(text).reply(true))`

**Comment by B-2U** [comment]
Hmm it should be good now, do i need to click something after mark the requests as resolved?

**Review by kangalio — APPROVED** [review]

**Pull request merged**